scripts/koji/koji_build.py

```diff
@@ -43,6 +52,53 @@ def koji_url(remote, hash):
         raise Exception("Unrecognized remote URL")
     return remote + "?#" + hash
 
+@contextmanager
+def local_branch(branch):
+    prev_branch = subprocess.check_output(['git', 'branch', '--show-current']).strip()
+    commit = subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode('utf8').strip()
+    subprocess.check_call(['git', 'checkout', '--quiet', commit])
+    try:
+        yield branch
+    finally:
+        subprocess.check_call(['git', 'checkout', prev_branch])
+
+def is_old_branch(b):
+    branch_time = datetime.strptime(b.split('/')[-1], TIME_FORMAT)
+    return branch_time < datetime.now() - timedelta(hours=3)
+
+def clean_old_branches(git_repo):
+    with cd(git_repo):
+        subprocess.check_call(['git', 'fetch'])
+        remote_branches = subprocess.check_output(['git', 'branch', '-rl', 'origin/koji/test/*/*']).decode('utf8').splitlines()
+        remote_branches = [b.strip()[len('origin/'):] for b in remote_branches]
+        old_branches = [b for b in remote_branches if is_old_branch(b)]
+        if old_branches:
+            print("removing outdated remote branch(es)", flush=True)
+            subprocess.check_call(['git', 'push', '--delete', 'origin'] + old_branches)
+
+def push_bumped_release(git_repo, trigram):
+    t = datetime.now().strftime(TIME_FORMAT)
+    branch = f'koji/test/{trigram}/{t}'
+    with cd(git_repo), local_branch(branch):
+        spec_paths = subprocess.check_output(['git', 'ls-files', 'SPECS/*.spec']).decode('utf8').splitlines()
+        assert len(spec_paths) == 1
+        spec_path = spec_paths[0]
+        with Specfile(spec_path) as spec:
+            # find the next build number
+            package = Path(spec_path).stem
+            builds = subprocess.check_output(['koji', 'list-builds', '--package', package]).decode('utf8').splitlines()[2:]
+            base_nvr = f'{package}-{spec.version}-{spec.release}.0.{trigram}.'
+            # use a regex to match %{macro} without actually expanding the macros
+            base_nvr_re = re.escape(re.sub('%{.+}', "@@@", base_nvr)).replace('@@@', '.*') + r'(\d+)'
+            build_matchs = [re.match(base_nvr_re, b) for b in builds]
```

Comment:
Nitpicking: `build_matches` looks more consistent with english spelling rules.

```suggestion
            build_matches = [re.match(base_nvr_re, b) for b in builds]
```